File: LeetCode/Java Solutions/Rearrange Array Elements by Sign/rearrange.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        int posIndex = 0, negIndex = 0;  // Pointers for filling positive and negative numbers
        
        // Separate and fill positive and negative numbers in the result array
        for (int num : nums) {
            if (num > 0) {
                result[posIndex * 2] = num;  // Place positive number
                posIndex++;
            } else {
                result[negIndex * 2 + 1] = num;  // Place negative number
                negIndex++;
            }
        }
        
        return result;
    }
}

File: LeetCode/Python Solutions/Rearrange Array Elements by Sign copy/rearrange.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        # Separate positive and negative numbers while preserving their order
        positive = [num for num in nums if num > 0]
        negative = [num for num in nums if num < 0]
        
        # Merge positive and negative numbers by alternating, starting with positive
        result = []
        for i in range(len(positive)):
            result.append(positive[i])
            result.append(negative[i])
        
        return result
